### ISSUE TITLE
Pressing the space bar while inside any TextEditor scrolls the window down

### ISSUE THREAD

**Issue by BinaryMuse** [atom/settings-view#767]
Pressing the space bar with the focus inside any settings-view TextEditor (to insert a space character) scrolls the window.

![spaces-pls](https://cloud.githubusercontent.com/assets/189606/14565036/07266e0a-02de-11e6-87fe-d51633aac082.gif)

**Comment by JSn1nj4** [comment]
What OS are you on, and what Atom version?

**Comment by BinaryMuse** [comment]
I'm on the latest master (1.9.0-dev-c4d4e0f), OS X 10.11.4.

**Comment by lee-dohm** [comment]
I could reproduce this on the Install tab, but not the Packages tab using Atom v1.9.0-dev-c4d4e0f on Mac OS X 10.11.4.

**Comment by jerone** [comment]
Reproducible on every tab (Settings, Keybindings, Packages, Themes & Install) on Windows 7 with Atom 1.8.0-beta2 and 1.8.0-beta3.

**Comment by NFiorentini** [comment]
Reproduced on every tab using Atom v1.8.0-beta4 on Windows 10.0.10586 Build 10586.

**Comment by muendelezaji** [comment]
Any updates on this?
Still happening on Atom 1.8.0 release, OS X 10.11.5. All tabs - Settings, Keybindings, Packages, Themes & Install

**Comment by leshniak** [comment]
I made a fix in my init.coffee, unfortunately works only partially. Matybe someone knows how to bind it better [[gist](https://gist.github.com/leshniak/355780d482fb9512bfc746c4b7ca50e6)]

**Comment by leshniak** [comment]
Ok, [new version](https://gist.github.com/leshniak/355780d482fb9512bfc746c4b7ca50e6) of my workaround works good. Put it in the init.coffee and enjoy.

**Comment by claudiofantacci** [comment]
@leshniak 's workaround works smoothly. Thanks!

Looking forward for a mainstream bugfix as well.

**Comment by muendelezaji** [comment]
Works for me too. Thanks @leshniak

**Comment by leshniak** [comment]
I think it should be fixed on atom-text-editor level, but maybe I'm wrong, I don't know Atom and Electron too much.

Btw, I made newer version of my workaround, w/o jQuery, it works with every atom-text-editor[mini] component. Link to gist in my previous comment.

**Comment by claudiofantacci** [comment]
@leshniak I opened an issue about this problem on atom and it was closed with the "wrong repo" label and redirected here. Thus I assume it's the correct one.

**Comment by NFiorentini** [comment]
Thank you, @leshniak!

**Comment by jerone** [comment]
I did some testing and I can reproduce this issue from at least 13 Sep 2015 7b60bb551b9db20e117e8b40167a16d8e96e4822 (anything earlier gives errors for now).

**Comment by jerone** [comment]
Identified the problem by removing the `textInput` event (in the Dev Tools) [attached](https://github.com/atom/atom/blob/896abe358c696db477605758a67d19bc63b8b1d8/src/text-editor-component.coffee#L246) on the `TextEditorComponent`.

**Comment by claudiofantacci** [comment]
Ok, then we might want to wait for a bugfix release and after that remove @leshniak workaround.

**Comment by ricefai** [comment]
Works for me, thanks

**Comment by bastilian** [comment]
Anyone working on this? Otherwise i would take a look at it.

**Comment by bastilian** [comment]
So, I've been looking into this, I believe the best way to address this would be to prevent this behaviour in the mini mode by using the [setMini](https://github.com/atom/atom/blob/master/src/text-editor.coffee#L592) function of TextEditor and add an event listener that handles the spacebar event, simply ignoring it and stop it from bubbling up and cause the the default chrome behaviour and scroll.

**BUT** this may just be a workaround and not the root of the problem, as I have yet to find out why this behaviour is not occur in places like find-and-replace, where TextEditor is used in mini as well.

My two theories are that it doesn't do it because it is observing changes with a handler that prevents this already or it does occur, but halts and is not effective since the view the TextEditor is in is not scrollable.

When I have found out why this doesn't happen in find-or-replace, I would open a issue on [atom/atom](/atom/atom) proposing the change and discuss it further there.

**Comment by DamnedScholar** [comment]
> I have yet to find out why this behaviour is not occur in places like find-and-replace, where TextEditor is used in mini as well.

You've made it so that find-and-replace scrolls in order to test this?

**Comment by bastilian** [comment]
@DamnedScholar no, that should read "...this behaviour does not occur..." :) and that is what I still need to figure out.

**Comment by ShadowLNC** [comment]
Reproducible on 1.8.0 release, all tabs (except Updates which has no text input).

In regards to find-and-replace, it usually either focuses the file TextEditor or one of the TextEditors (mini) inside the find-and-replace dialog, but on the occasions that I get all cursors to disappear, spacebar will refocus the file TextEditor and insert the space there (suggesting the file TextEditor was actually focused without a cursor).

Clicking on empty space in the tab bar will also unfocus the cursor, but the spacebar then has no effect (suggesting the file TextEditor is properly unfocused this time).

Spacebar in tile-view also scrolls, though that may be intended behaviour (and it's not this package, but it's worth noting the behaviour).

**Comment by bastilian** [comment]
I had to modify the style of the elements a bit to make it happen, but as I suspected this behaviour happens not only in settings-view:
![find-replace](https://cloud.githubusercontent.com/assets/7757/16480218/22d26ab6-3ea5-11e6-8634-bccc2eb7d021.gif)

I'll investigate this more what triggers the behaviour, how it can be fixed and open an issue on atom/atom. I'm not really happy with the solution I proposed above, since I have the feeling there is a tiny bit more behind it.

**Comment by jerone** [comment]
Remember there are multiple keys triggering this unintended behavior:

* <kbd>SpaceBar</kbd>
* <kbd>ArrowDown</kbd>
* <kbd>ArrowUp</kbd>
* <kbd>PgUp</kbd>
* <kbd>PgDn</kbd>

**Comment by bastilian** [comment]
Oh, thanks for noting that! I'll include these in the coming issue on atom as well.

**Comment by mbigras** [comment]
I'm having this same problem with 
OSX 10.11.5
Atom    : 1.10.0-dev-04ff1b5
Electron: 0.37.8
Chrome  : 49.0.2623.75
Node    : 5.10.0

@leshniak's script saves the day 😄

It can be found above but for anyone who scrolled to the bottom, here is the [gist](https://gist.github.com/leshniak/f3b163b23bbfac5dad313b628f3c09ac). Just append it to your `init.coffee` script which can be found at: `/Users/<name>/.atom/init.coffee` on OSX

**Comment by andrewstart** [comment]
For your coming issue on atom: I've been seeing this problem in the custom editors that I am working on. When any atom-text-editor (mini or not) is contained within a scrollable section of the page, then spacebar causes that section to scroll.

**Comment by philxchen** [comment]
Reproduced.

Windows 10 build 14366 & 14361
Atom 1.9.0-beta0

**Comment by danielbayley** [comment]
😞
I'm also getting this on:

Atom 1.8.0
OS X 10.11.6
settings-view 0.235.1

**Comment by refactorized** [comment]
Also getting this in 1.8.0 as well as strange config values which have extra spaces between numbers ( 300 becomes 3 0 0) and doubled spaces ( two spaces where one should be)

**Comment by xHN35RQ** [comment]
Also happens on ubuntu 16.04 

```
Atom    : 1.8.0
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

**Comment by andyrichardson** [comment]
Just encountered this on Fedora 24

**Comment by damieng** [comment]
I don't think we need any more messages about it happening in place. It happens everywhere on all systems.

**Comment by nathansobo** [comment]
This just came to my attention and is a really bad regression. My apologies to everyone affected. A fix will be out with the next release (shortly).

**Comment by sharifzadesina** [comment]
didn't fixed...

ubuntu 16.04
atom 1.8

**Comment by brainscript** [comment]
Same here. :disappointed:

[eOS Freya](http://elementary.io) (Ubuntu 14.04)
Atom 1.8.0

**Comment by jerone** [comment]
I can however confirm this being fixed for Windows.

**Comment by mnquintana** [comment]
This has been fixed for the *next* release of Atom, which isn't out yet. If you want to get the fix sooner, you might be interested in the [Beta channel](https://atom.io/beta).

**Comment by Kaukov** [comment]
I can confirm this has been fixed in 1.9.0

Tested on Windows 10 Pro x64
